Add unpublishing details to document export

leenagupte opened the pull request:
Trello: https://trello.com/c/8U7JPIHT

## What's changed?
Adds unpublishing details to the export.

## Why?
So the that information it contains, for example the reason the document was unpublished, the user facing explanation, and where users may be being redirected to are not lost when the document is exported.

## Expected changes
![Screenshot 2019-11-06 at 12 21 30](https://user-images.githubusercontent.com/5793815/68297942-0fe24880-0090-11ea-8228-10c07ba029a0.png)

ChrisBAshton commented on app/presenters/document_export_presenter.rb:
```diff
@@ -141,6 +142,20 @@ def present_translations(edition)
+          id
+          explanation
+          alternative_url
+          redirect created_at
```
Looks like these two slipped in on the same line.

ChrisBAshton commented on test/unit/presenters/document_export_presenter_test.rb:
```diff
@@ -393,6 +393,22 @@ class DocumentExportPresenterTest < ActiveSupport::TestCase
+    }
+
+    assert_equal expected, result.dig(:editions, 0, :unpublishing)
+  end
```
Is it worth writing another test for unpublish-and-redirect just to make sure `redirect` and `alternative_url` are set?

ChrisBAshton commented on app/presenters/document_export_presenter.rb:
```diff
@@ -143,17 +143,22 @@ def present_translations(edition)
+            updated_at
+          ],
+        )
+        .tap do |unpublishing|
```
This feels cleaner than tapping:

```ruby
.merge(reason: edition.unpublishing.unpublishing_reason.name)
```

leenagupte commented:
@ChrisBAshton Thanks for your comments, I've addressed them and fixed up.

> I wonder if explanation and reason are confusingly similar? AFAIK the explanation is user facing, and the reason is an internal note. Perhaps user_facing_explanation and unpublishing_reason are better names?

I've updated `reason` to be `unpublishing_reason` as suggested, but I've kept `explanation` as both fields now match the original column names.
